R/prepare-admin-panel.R

```diff
@@ -279,81 +319,96 @@ prepare_admin_panel_components <- function(
     plot_daily_stats(per_day_plot_data())
   })
 
-  shiny::observe({
-    if (length(time_daily()) > 0) {
-      output$total_time <- semantic.dashboard::renderValueBox({
-        time_hours <- time_daily() %>%
-          dplyr::pull(.data$time_spent) %>%
-          mean() %>%
-          convert_timediff_to_hm()
-
-        semantic.dashboard::valueBox(
-          value = time_hours,
-          subtitle = "Average time spent daily",
-          icon = semantic.dashboard::icon("User Circle"),
-          color = "yellow",
-          width = 16
-        )
-      })
+  output$total_time <- semantic.dashboard::renderValueBox({
+    if (is_log_empty()) {
+      time_hours <- convert_timediff_to_hm(0)
     } else {
-      NULL
+      time_hours <- time_daily() %>%
+        dplyr::pull(.data$time_spent) %>%
+        mean() %>%
+        convert_timediff_to_hm()
     }
+
+    semantic.dashboard::valueBox(
+      value = time_hours,
+      subtitle = "Average time spent daily",
+      icon = semantic.dashboard::icon("User Circle"),
+      color = "yellow",
+      width = 16
+    )
   })
 
-  shiny::observe({
-    if (nrow(selected_log_data()) > 0) {
-      output$total_users <- semantic.dashboard::renderValueBox({
-        semantic.dashboard::valueBox(
-          value = length(unique(
-            selected_log_data() %>%
-              dplyr::filter(.data$type == "login") %>%
-              dplyr::filter(.data$username != "") %>%
-              dplyr::pull(.data$username)
-          )),
-          subtitle = "Unique users (with logins)",
-          icon = semantic.dashboard::icon("User Circle"),
-          color = "red",
-          width = 16
-        )
-      })
+  output$total_users <- semantic.dashboard::renderValueBox({
+    if (is_log_empty()) {
```

Comment:
I like the approach of applying the same logic to all reactive, but I wonder if this is really necessary as `length(character(0L))` is 0

This also applies to others that follow this pattern, `reactive_call() %>% filter(...) %>% filter(...) %>% pull(...)`

Reply:
You are right. The check for `is_log_empty()` is necessary for two reactives only.
It was added to other reactives for consistency. I'll remove the ones that are not required.

Reply:
I don't think there's a right or wrong here. I tend to try to avoid unnecessary code and go on a case by case.

Either way is fine by me, although I see you already made the change :+1: